state/upgrades.go

```diff
@@ -481,6 +482,80 @@ func updateLegacyLXDCredentialsOps(st *State, cred cloud.Credential) ([]txn.Op,
 	return ops, nil
 }
 
+// UpdateLegacyKubernetesCloudCredentials updates the cloud credentials for
+// Kubernetes clouds. This is to introduce bug fix changes from 2.9 onwards.
+func UpdateLegacyKubernetesCloudCredentials(st *State) error {
+	ops, err := updateLegacyKubernetesCloudAuthTypesOps(st)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return st.db().RunTransaction(ops)
+}
+
+func updateLegacyKubernetesCloudAuthTypesOps(st *State) ([]txn.Op, error) {
```

Comment:
this does more than just update the other types
perhaps just call it updateLegacyKubernetesCredentialsOps() and rename then current updateLegacyKubernetesCredentialsOps to updateLegacyKubernetesCredentialValuesOps or something

Reply:
Done & Done